### ISSUE TITLE
Need some help.

### ISSUE THREAD

**Issue by AntwanR942** [onlaj/Piano-LED-Visualizer#218]
To cut a long story short I had this working but due to the way I positioned it the wires snapped off the LED strip so i purchased a new strip connected it up and nothing is happening. I completely reinstalled the OS and the visualizer back onto the RPI but still nothing. I tried selecting an animation but nothing happens when i click the button. Help would be greatly appreciated. (I have taken some photos, please check them about to see how everything is wired and if you can see anything wrong.)

![image1](https://user-images.githubusercontent.com/46630572/113452897-e05a7b80-93fc-11eb-9c28-e4293c8a3ff6.jpg)
![image2](https://user-images.githubusercontent.com/46630572/113452902-e18ba880-93fc-11eb-8a65-e51dbad41f0c.jpg)
![image3](https://user-images.githubusercontent.com/46630572/113452904-e2bcd580-93fc-11eb-9ec1-7cafa09444fe.jpg)

**Comment by AntwanR942** [comment]
I would be willing to replace the RPI but only if someone can assure me that it is the controller that is broken. I am thinking this because randomly the (~7 leds) lights all went white but still not responding to commands. From what I've seen this is a "timing" issue where the WS2812B strip requires very specific timing and thus isn't responding as expected but it also may be a faulty connection somewhere.

**Comment by AntwanR942** [comment]
Also, I could measure 5V on the LED strip.

**Comment by AntwanR942** [comment]
I've ran some tests in the tests directory and measured some values while they were running. I could still measure 5V across the LED strip as usual. I could measure a fluctuation of voltage from 0V to ~2V from the 2 GPIO pins that are connect to the strip however there was no voltage across the middle and negative contacts actually on the strip. Does this suggest that a faulty connection is not allowing voltage to get to the strip or is there meant to be nothing across the middle(data) negative contacts?

**Comment by bobsmith1950** [comment]
I've just started my LED project and had lots and lots of issues to start with.

Now it's behaving itself - I added a resistor on the data line (~470 ohm) and a capacitor across the power connector + / - terminals (1000uF - 6.3V to 25V) - this has cleaned up a lot of my problems.

Also the connections to GND and PIN18 (data signal) on the PI Zero can easily break/short - so i added a GPIO expander to give me space to use jumpers instead.
